### ISSUE TITLE
PSP-Net Inplace modification of a variable with requires_grad=True 

### ISSUE THREAD

**Issue by tedyhabtegebrial** [meetshah1995/pytorch-semseg#121]
Can't backward with pspnet, because of an in-place modified variable. When I change the model segnet it works.

**Comment by adam9500370** [comment]
Hi, @tedyhabtegebrial .
You can see #96 .

**Comment by tedyhabtegebrial** [comment]
Thanks! @adam9500370 

Using Dropout inplace causes the problem

Just for reference, the problem is solved by changing
```python
self.dropout = nn.Dropout2d(p=0.1, inplace=True)
```
```python
self.dropout = nn.Dropout2d(p=0.1, inplace=False)
```
